### PR TITLE
refactor(HomeFeed): code

### LINKED ISSUE #365: home-feed
[Epic 356](https://github.com/haqq-network/haqq-wallet/issues/356)

### PR DESCRIPTION
**Pull request by samandarkuchkarov**
Fixes #365

## Proposed Changes

Change the imports to absolute:
Move text to i18n
Change the colors from the constants to similar ones in Color
Transferring the screen logic to components
Replace navigation hooks with typed navigation hooks in folder src/hooks
Screenshot 2022-12-09 at 10 30 03

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: src/components/wallet-card.tsx [hunk 1]
```diff
@@ -83,64 +68,67 @@ export const WalletCard = ({address}: BalanceProps) => {
...
+        <CopyButton style={styles.copyButton} value={wallet.address}>
+          <Text t14 color={Color.textBase3}>
             {formattedAddress}
           </Text>
-          <Copy color={LIGHT_GRAPHIC_BASE_3} style={page.marginLeft} />
+          <Icon
+            name="copy"
+            color={Color.graphicBase3}
+            style={styles.marginLeft}
+          />
```
Comment (gHashTag) [inline src/components/wallet-card.tsx:92]: Determine the size of the icon (`i<N>` prop)

File: src/components/wallet-card.tsx [hunk 2]
```diff
@@ -83,64 +68,67 @@ export const WalletCard = ({address}: BalanceProps) => {
...
-        <View style={page.button}>
+        <View style={styles.button}>
           {IS_IOS && <BlurView action="receive" cardState={cardState} />}
-          <IconButton style={page.spacer} onPress={onPressQR}>
-            <ArrowReceive color={LIGHT_GRAPHIC_BASE_3} />
-            <Text clean style={page.buttonText}>
-              Receive
-            </Text>
+          <IconButton style={styles.spacer} onPress={onPressQR}>
+            <Icon name="arrow_receive" color={Color.graphicBase3} />
```
Comment (gHashTag) [inline src/components/wallet-card.tsx:119]: Determine the size of the icon using the prop

File: src/components/wallet-card.tsx [hunk 3]
```diff
@@ -83,64 +68,67 @@ export const WalletCard = ({address}: BalanceProps) => {
...
+      <View style={styles.buttonsContainer}>
+        <View style={styles.button}>
           {IS_IOS && <BlurView action="sent" cardState={cardState} />}
-          <IconButton style={page.spacer} onPress={onPressSend}>
-            <ArrowSend color={LIGHT_GRAPHIC_BASE_3} />
-            <Text clean style={page.buttonText}>
-              Send
-            </Text>
+          <IconButton style={styles.spacer} onPress={onPressSend}>
+            <Icon name="arrow_send" color={Color.graphicBase3} />
```
Comment (gHashTag) [inline src/components/wallet-card.tsx:112]: Determine the size of the icon using the prop

File: src/components/transaction-empty.tsx [hunk 4]
```diff
@@ -2,22 +2,20 @@ import React from 'react';
...
 
 export const TransactionEmpty = () => {
   return (
-    <View style={page.container}>
-      <NoTransactionsIcon color={LIGHT_GRAPHIC_SECOND_3} style={page.space} />
-      <Text t14 style={{color: LIGHT_TEXT_SECOND_1}}>
-        No transactions
-      </Text>
+    <View style={styles.container}>
+      <NoTransactionsIcon color={Color.graphicSecond3} style={styles.space} />
```
Comment (gHashTag) [inline src/components/transaction-empty.tsx:12]: `NoTransactionsIcon` not handle Color type inside, use `getColor()`

**Review by gHashTag — APPROVED** [review]
Fix comments

File: src/components/transaction-empty.tsx (hunk 4 above)
Comment (samandarkuchkarov) [reply src/components/transaction-empty.tsx]: done

File: src/components/wallet-card.tsx (hunk 3 above)
Comment (samandarkuchkarov) [reply src/components/wallet-card.tsx]: done

File: src/components/wallet-card.tsx (hunk 1 above)
Comment (samandarkuchkarov) [reply src/components/wallet-card.tsx:93]: done

File: src/components/wallet-card.tsx (hunk 2 above)
Comment (samandarkuchkarov) [reply src/components/wallet-card.tsx]: done

**Review by gHashTag — APPROVED** [review]

File: src/components/transactions/send.tsx [hunk 5]
```diff
@@ -20,14 +20,15 @@ export const TransactionSend = ({item, onPress}: TransactionPreviewProps) => {
...
+      <View style={styles.container}>
+        <View style={styles.iconWrapper}>
           <Icon name="arrow_send" color={Color.graphicBase1} />
         </View>
         <DataContent
-          style={page.infoContainer}
-          title="Sent"
-          subtitle={`to ${shortAddress(item.to, '•')}`}
+          style={styles.infoContainer}
+          titleI18n={I18N.sent}
```
Comment (vivalaakam) [inline src/components/transactions/send.tsx:29]: I18N.transactionSendTitle

File: src/components/transactions/send.tsx [hunk 6]
```diff
@@ -20,14 +20,15 @@ export const TransactionSend = ({item, onPress}: TransactionPreviewProps) => {
...
+          subtitleI18nParams={{address: shortAddress(item.to, '•')}}
+          subtitleI18n={I18N.to}
```
Comment (vivalaakam) [inline src/components/transactions/send.tsx:31]: I18N.transactionSendTo

File: src/i18n.ts [hunk 7]
```diff
@@ -325,6 +319,7 @@ const en: Record<I18N, string> = {
   [I18N.transactionAddressPlaceholder]: 'Address (0x) or contact name',
   [I18N.transactionAddressError]: 'Incorrect address',
   [I18N.transactionAddressLabel]: 'Send to',
+  [I18N.TransactionEmpty]: 'No transactions',
```
Comment (vivalaakam) [inline src/i18n.ts:322]: I18N.transactionEmpty

File: src/components/home-feed.tsx [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+  useEffect(() => {
+    transactions.on('transactions', onTransactionList);
+    wallets.on('backupMnemonic', onBackupMnemonic);
+    user.on('change', onTransactionList);
+    return () => {
+      transactions.off('transactions', onTransactionList);
+      wallets.off('backupMnemonic', onBackupMnemonic);
+      user.off('change', onTransactionList);
+    };
+  }, [onBackupMnemonic, onTransactionList, transactions, user, wallets]);
```
Comment (vivalaakam) [inline src/components/home-feed.tsx:92]: move on screen pls

File: src/components/home-feed.tsx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    const actions = wallets.addressList.map(address =>
+      transactions.loadTransactionsFromExplorer(address),
+    );
+
+    actions.push(wallets.checkBalance());
+
+    Promise.all(actions).then(() => {
+      setRefreshing(false);
+    });
+  }, [transactions, wallets]);
```
Comment (vivalaakam) [inline src/components/home-feed.tsx:56]: move on screen pls

File: src/components/home-feed.tsx (hunk 8 above)
Comment (samandarkuchkarov) [reply src/components/home-feed.tsx]: done

File: src/i18n.ts (hunk 7 above)
Comment (samandarkuchkarov) [reply src/i18n.ts]: done

File: src/components/transactions/send.tsx (hunk 5 above)
Comment (samandarkuchkarov) [reply src/components/transactions/send.tsx]: done

File: src/components/home-feed.tsx (hunk 9 above)
Comment (samandarkuchkarov) [reply src/components/home-feed.tsx]: done

**Review by vivalaakam — APPROVED** [review]

**Pull request merged**